### PR TITLE
Fixes frozen string exception when checking equivalent gemfile and lockfile

### LINKED ISSUE #4520: can't modify frozen string
possibly related to #4459. Happening again in 1.12.

- What did you do?

  I ran the command `/usr/local/bin/bundle install --path /var/www/.../shared/bundle --without development test --deployment --quiet`

- What did you expect to happen?

  Install the gems. This has worked previously on this host.

- What happened instead?

  error

Error details

    RuntimeError: can't modify frozen String
      /var/lib/gems/2.3.0/gems/bundler-1.12.1/lib/bundler/definition.rb:432:in `pretty_dep'
      /var/lib/gems/2.3.0/gems/bundler-1.12.1/lib/bundler/definition.rb:374:in `block in ensure_equivalent_gemfile_and_lockfile'
      /var/lib/gems/2.3.0/gems/bundler-1.12.1/lib/bundler/definition.rb:374:in `map'
      /var/lib/gems/2.3.0/gems/bundler-1.12.1/lib/bundler/definition.rb:374:in `ensure_equivalent_gemfile_and_lockfile'
      /var/lib/gems/2.3.0/gems/bundler-1.12.1/lib/bundler/installer.rb:61:in `run'
      /var/lib/gems/2.3.0/gems/bundler-1.12.1/lib/bundler/installer.rb:22:in `install'
      /var/lib/gems/2.3.0/gems/bundler-1.12.1/lib/bundler/cli/install.rb:102:in `run'
      /var/lib/gems/2.3.0/gems/bundler-1.12.1/lib/bundler/cli.rb:175:in `install'
      /var/lib/gems/2.3.0/gems/bundler-1.12.1/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
      /var/lib/gems/2.3.0/gems/bundler-1.12.1/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
      /var/lib/gems/2.3.0/gems/bundler-1.12.1/lib/bundler/vendor/thor/lib/thor.rb:359:in `dispatch'
      /var/lib/gems/2.3.0/gems/bundler-1.12.1/lib/bundler/vendor/thor/lib/thor/base.rb:440:in `start'
      /var/lib/gems/2.3.0/gems/bundler-1.12.1/lib/bundler/cli.rb:11:in `start'
      /var/lib/gems/2.3.0/gems/bundler-1.12.1/exe/bundle:27:in `block in <top (required)>'
      /var/lib/gems/2.3.0/gems/bundler-1.12.1/lib/bundler/friendly_errors.rb:98:in `with_friendly_errors'
      /var/lib/gems/2.3.0/gems/bundler-1.12.1/exe/bundle:19:in `<top (required)>'
      /usr/local/bin/bundle:23:in `load'
      /usr/local/bin/bundle:23:in `<main>'

Environment

  Bundler   1.12.1
  Rubygems  2.5.1
  Ruby      2.3.0p0 (2015-12-25 revision 0) [x86_64-linux-gnu]
  GEM_HOME  /var/www/.../shared/bundle/ruby/2.3.0
  GEM_PATH
  Git       2.7.4

      Bundler settings

  frozen
    Set for your local app (/var/www/.../releases/20160503152753/.bundle/config): "1"
  path
    Set for your local app (/var/www/.../releases/20160503152753/.bundle/config): "/var/www/.../shared/bundle"
  without
    Set for your local app (/var/www/.../releases/20160503152753/.bundle/config): "development:test"
  disable_shared_gems
    Set for your local app (/var/www/.../releases/20160503152753/.bundle/config): true
  orig_path
    Set via BUNDLE_ORIG_PATH: "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games"

### PR DESCRIPTION
**Pull request by grzuy**
closes #4520

### CODE REVIEW

File: lib/bundler/definition.rb [hunk 1]
```diff
@@ -434,7 +434,7 @@ def nothing_changed?
     end
 
     def pretty_dep(dep, source = false)
-      msg = "#{dep.name}"
+      msg = String.new("#{dep.name}")
```
Comment (segiddins) [inline lib/bundler/definition.rb]: The interpolation and the strong literal are unnecessary now

Comment (grzuy) [inline lib/bundler/definition.rb]: Nice catch, thanks!

**Comment by segiddins** [comment]
👍🏻

**Comment by segiddins** [comment]
@homu r+

File: lib/bundler/definition.rb [hunk 2]
```diff
@@ -434,7 +434,7 @@ def nothing_changed?
     end
 
     def pretty_dep(dep, source = false)
-      msg = "#{dep.name}"
+      msg = String.new(dep.name)
```
Comment (ixti) [inline lib/bundler/definition.rb]: `dep.name.dup` would look better and would be more than enough :D

**Pull request merged**